pyproject.toml

```diff
@@ -1,50 +1,57 @@
-[tool.poetry]
+[project]
 name = "linkml-map"
 version = "0.0.0"
 description = "a framework for specifying and executing mappings between data models"
-authors = ["cmungall <cjm@berkeleybop.org>"]
 readme = "README.md"
+authors = [
+    {name = "cmungall", email = "cjm@berkeleybop.org"},
+]
+requires-python = "<=3.13,>=3.9"
 
-[tool.poetry.dependencies]
-python = "^3.9"
-asteval = "^0"
-click = "^8"
-curies = "*"
-duckdb = { version = "^0", optional = true }
-graphviz = { version = "^0.20.1", optional = true }
-jinja2 = "^3"
-lark = { version = "^1", optional = true }
-linkml-runtime = ">=1.7.2"
-pint = { version = "^0", optional = true }
-pydantic = ">=2.0.0"
-pyyaml = "*"
-ucumvert = { version = "^0", optional = true }
+dependencies = [
+    "asteval<1,>=0",
+    "click<9,>=8",
+    "curies",
+    "jinja2<4,>=3",
+    "linkml-runtime>=1.7.2",
+    "pydantic>=2.0.0",
+    "pyyaml",
+]
 
-[tool.poetry.extras]
-duckdb = ["duckdb"]
-graphviz = ["graphviz"]
-units = ["lark", "pint", "ucumvert"]
+[project.optional-dependencies]
+duckdb = [
+    "duckdb<1,>=0",
+]
+graphviz = [
+    "graphviz<1.0.0,>=0.20.1",
+]
+units = [
+    "lark<2,>=1",
+    "pint<1,>=0",
+    "ucumvert<1,>=0",
+]
+dev = [
+    "deepdiff<7.0.0,>=6.7.1",
+    "deptry<1.0.0,>=0.21.1",
+    "jupyter<2.0.0,>=1.0.0",
+    "linkml>=1.7.0",
+    "mkdocs-mermaid2-plugin<1.0.0,>=0.6.0",
+    "mkdocs-windmill",
+    "mkdocstrings[crystal,python]",
+    "mknotebooks<1.0.0,>=0.8.0",
+    "pytest>=7.3.1",
+    "ruff>=0.11.2",
+    "tox>=4.24.2",
+]
 
-[tool.poetry.group.dev.dependencies]
-deepdiff = "^6.7.1"
-deptry = "^0.21.1"
-jupyter = "^1.0.0"
-linkml = ">=1.7.0"
-mkdocs-mermaid2-plugin = "^0.6.0"
-mkdocs-windmill = "*"
-mkdocstrings = {extras = ["crystal", "python"], version = "*"}
-mknotebooks = "^0.8.0"
-pytest = "^7.3.1"
-tox = "*"
+[project.scripts]
+linkml-map = "linkml_map.cli.cli:main"
```

Comment:
After this add a section to set the default version for dynamic versioning and require poetry>2:
```
[tool.poetry]
requires-poetry = ">=2.0"
version = "0.0.0"

[tool.poetry.requires-plugins]
poetry-dynamic-versioning = { version = ">=1.0.0,<2.0.0", extras = ["plugin"] }
```

Reply:
Added below